### DISCUSSION TITLE
can't use this project in sub directory

### DISCUSSION THREAD

**Discussion by nitiphone** [firefly-iii/firefly-iii#6021] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I installed the project on sub directory of my hosting.
www.example/firefly/
after run composer update, migrate and db seed
when I access the website it's say message

Danger! This directory should not be open to the public!
/public/ should be the document root of your web server.
Leaving your web server configured like this is a huge security risk.
Please [read more on the Github help pages](https://github.com/firefly-iii/help/wiki/Configure-your-webserver-correctly).

after that it's redirect to www.example/firefly/public/login but show message error 500

### Expected behaviour

normally is should work on register page and my localhost

### Debug information

/storage/log
[2022-04-22T16:20:30.080405+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:20:37.205634+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:20:39.238301+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:20:59.383186+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:22:15.653009+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:22:16.343837+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:22:16.652389+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:22:17.013543+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:27:09.677362+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:29:18.567960+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:34:41.743512+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:34:42.164195+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:40:39.689845+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []
[2022-04-22T16:46:20.151697+07:00] local.INFO: AUDIT: Show login form (1.1). (183.182.111.114 -> GET:https://dev.bdb.com.la/firefly/public/login) [] []

**Comment by JC5** [comment]
Unfortunately the logs you've shared do not show the root cause for the 500 error.

**Reply by nitiphone** [reply]
Error log message

> `
Next InvalidArgumentException: Error loading /home/devbdbcom/public_html/firefly/resources/views/errors/500.twig: Unable to create the cache directory (/home/devbdbcom/public_html/firefly/storage/framework/views/twig/61). in /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Compiler.php:100
Stack trace:
#0 /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Twig.php(92): TwigBridge\Engine\Compiler->load('/home/devbdbcom...')
#1 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(139): TwigBridge\Engine\Twig->get('/home/devbdbcom...', Array)
#2 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(122): Illuminate\View\View->getContents()
#3 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(91): Illuminate\View\View->renderContents()
#4 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Http/Response.php(69): Illuminate\View\View->render()
#5 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Http/Response.php(35): Illuminate\Http\Response->setContent(Object(Illuminate\View\View))
#6 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(57): Illuminate\Http\Response->__construct(Object(Illuminate\View\View), 500, Array)
#7 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(87): Illuminate\Routing\ResponseFactory->make(Object(Illuminate\View\View), 500, Array)
#8 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(613): Illuminate\Routing\ResponseFactory->view('errors::500', Array, 500, Array)
#9 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(538): Illuminate\Foundation\Exceptions\Handler->renderHttpException(Object(Symfony\Component\HttpKernel\Exception\HttpException))
#10 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(444): Illuminate\Foundation\Exceptions\Handler->prepareResponse(Object(Illuminate\Http\Request), Object(Symfony\Component\HttpKernel\Exception\HttpException))
#11 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(364): Illuminate\Foundation\Exceptions\Handler->renderExceptionResponse(Object(Illuminate\Http\Request), Object(InvalidArgumentException))
#12 /home/devbdbcom/public_html/firefly/app/Exceptions/Handler.php(128): Illuminate\Foundation\Exceptions\Handler->render(Object(Illuminate\Http\Request), Object(InvalidArgumentException))
#13 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): FireflyIII\Exceptions\Handler->render(Object(Illuminate\Http\Request), Object(InvalidArgumentException))
#14 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(173): Illuminate\Foundation\Bootstrap\HandleExceptions->renderHttpResponse(Object(InvalidArgumentException))
#15 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(231): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(InvalidArgumentException))
#16 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}(Object(InvalidArgumentException))
#17 {main}
  thrown {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Uncaught RuntimeException: Unable to create the cache directory (/home/devbdbcom/public_html/firefly/storage/framework/views/twig/61). in /home/devbdbcom/public_html/firefly/vendor/twig/twig/src/Cache/FilesystemCache.php:53
Stack trace:
#0 /home/devbdbcom/public_html/firefly/vendor/twig/twig/src/Environment.php(349): Twig\\Cache\\FilesystemCache->write('/home/devbdbcom...', '<?php\
\
use Twig...')
#1 /home/devbdbcom/public_html/firefly/vendor/twig/twig/src/Environment.php(309): Twig\\Environment->loadTemplate('__TwigTemplate_...', '/home/devbdbcom...')
#2 /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Compiler.php(98): Twig\\Environment->load('/home/devbdbcom...')
#3 /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Twig.php(92): TwigBridge\\Engine\\Compiler->load('/home/devbdbcom...')
#4 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(139): TwigBridge\\Engine\\Twig->get('/home/devbdbcom...', Array)
#5 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(122): Illuminate\\View\\View->getContents()
#6 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(91): Illuminate\\View\\View->renderContents()
#7 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Http/Response.php(69): Illuminate\\View\\View->render()
#8 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Http/Response.php(35): Illuminate\\Http\\Response->setContent(Object(Illuminate\\View\\View))
#9 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(57): Illuminate\\Http\\Response->__construct(Object(Illuminate\\View\\View), 500, Array)
#10 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(87): Illuminate\\Routing\\ResponseFactory->make(Object(Illuminate\\View\\View), 500, Array)
#11 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(613): Illuminate\\Routing\\ResponseFactory->view('errors::500', Array, 500, Array)
#12 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(538): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException(Object(Symfony\\Component\\HttpKernel\\Exception\\HttpException))
#13 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(444): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse(Object(Illuminate\\Http\\Request), Object(Symfony\\Component\\HttpKernel\\Exception\\HttpException))
#14 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(364): Illuminate\\Foundation\\Exceptions\\Handler->renderExceptionResponse(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#15 /home/devbdbcom/public_html/firefly/app/Exceptions/Handler.php(128): Illuminate\\Foundation\\Exceptions\\Handler->render(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#16 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): FireflyIII\\Exceptions\\Handler->render(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#17 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(173): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse(Object(InvalidArgumentException))
#18 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(231): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException(Object(InvalidArgumentException))
#19 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}(Object(InvalidArgumentException))
#20 {main}

Next InvalidArgumentException: Error loading /home/devbdbcom/public_html/firefly/resources/views/errors/500.twig: Unable to create the cache directory (/home/devbdbcom/public_html/firefly/storage/framework/views/twig/61). in /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Compiler.php:100
Stack trace:
#0 /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Twig.php(92): TwigBridge\\Engine\\Compiler->load('/home/devbdbcom...')
#1 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(139): TwigBridge\\Engine\\Twig->get('/home/devbdbcom...', Array)
#2 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(122): Illuminate\\View\\View->getContents()
#3 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/View/View.php(91): Illuminate\\View\\View->renderContents()
#4 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Http/Response.php(69): Illuminate\\View\\View->render()
#5 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Http/Response.php(35): Illuminate\\Http\\Response->setContent(Object(Illuminate\\View\\View))
#6 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(57): Illuminate\\Http\\Response->__construct(Object(Illuminate\\View\\View), 500, Array)
#7 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(87): Illuminate\\Routing\\ResponseFactory->make(Object(Illuminate\\View\\View), 500, Array)
#8 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(613): Illuminate\\Routing\\ResponseFactory->view('errors::500', Array, 500, Array)
#9 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(538): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException(Object(Symfony\\Component\\HttpKernel\\Exception\\HttpException))
#10 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(444): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse(Object(Illuminate\\Http\\Request), Object(Symfony\\Component\\HttpKernel\\Exception\\HttpException))
#11 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(364): Illuminate\\Foundation\\Exceptions\\Handler->renderExceptionResponse(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#12 /home/devbdbcom/public_html/firefly/app/Exceptions/Handler.php(128): Illuminate\\Foundation\\Exceptions\\Handler->render(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#13 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): FireflyIII\\Exceptions\\Handler->render(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#14 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(173): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse(Object(InvalidArgumentException))
#15 /home/devbdbcom/public_html/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(231): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException(Object(InvalidArgumentException))
#16 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}(Object(InvalidArgumentException))
#17 {main}
  thrown at /home/devbdbcom/public_html/firefly/vendor/rcrowe/twigbridge/src/Engine/Compiler.php:100)
[stacktrace]
#0 {main}
"} `

**Reply by JC5** [reply]
That's a local permissions error, not a Firefly III error. Make sure storage directory is writable.
